src/components/Modal.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+type Props = { isActive: boolean; onClick: () => void };
+
+class Modal extends React.Component<Props> {
+    render() {
+        const isActive = this.props.isActive;
```

Comment:
The render method shouldn't have any extra logic in here. 

You should be defining a constructor instead that initializes the props

```
class Modal extends React.Component<IModalProps>
    constructor(props: IModalProps) {
        super(props);
    }
```

It may seem redundant now but when we bind these callbacks, this is where we will do it.